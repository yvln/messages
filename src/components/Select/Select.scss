@import "styles/_variables.scss";

.Select {
  margin-right: $spacing-medium;
  
  select, option {
    border: none;
    height: 30px;
    width: 70px;
  }
}