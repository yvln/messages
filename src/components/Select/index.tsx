import * as React from 'react';

import './Select.scss';

interface IOption {
  label: string;
  value: string;
}

interface IInjectedProps {
  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
  options?: IOption[];
}

interface IDefaultProps {
  options: IOption[];
}

export type IProps = IInjectedProps & IDefaultProps;

export default class Select extends React.Component<IProps> {
  static defaultProps = {
    options: [
      {
        label: 'Public',
        value: 'public',
      },
      {
        label: 'Private',
        value: 'private',
      },
    ],
  };

  renderOption = (option: IOption) => {
    return (
      <option key={option.value} value={option.value}>
        {option.label}
      </option>
    );
  };

  render() {
    return (
      <div className="Select">
        <select onChange={this.props.onChange}>
          {this.props.options.map(this.renderOption)}
        </select>
      </div>
    );
  }
}
