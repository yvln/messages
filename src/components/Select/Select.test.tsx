import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import Select, { IProps } from '.';

configure({ adapter: new Adapter() });

describe('src/components/Select', () => {
  let props: IProps;

  beforeEach(() => {
    props = {
      onChange: jest.fn(),
      options: [
        {
          label: 'Public',
          value: 'public',
        },
        {
          label: 'Private',
          value: 'private',
        },
      ],
    };
  });

  it('should render', () => {
    const component = shallow(<Select {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should call "onChange" when changing its value', () => {
    const component = mount(<Select {...props} />);
    component.find('select').simulate('change');
    expect(props.onChange).toBeCalled();
  });
});
