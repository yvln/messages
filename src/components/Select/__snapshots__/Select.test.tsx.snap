// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Select should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "Public",
          "value": "public",
        },
        Object {
          "label": "Private",
          "value": "private",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[MockFunction]}
      >
        <option
          value="public"
        >
          Public
        </option>
        <option
          value="private"
        >
          Private
        </option>
      </select>,
      "className": "Select",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="public"
          >
            Public
          </option>,
          <option
            value="private"
          >
            Private
          </option>,
        ],
        "onChange": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "public",
          "nodeType": "host",
          "props": Object {
            "children": "Public",
            "value": "public",
          },
          "ref": null,
          "rendered": "Public",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "private",
          "nodeType": "host",
          "props": Object {
            "children": "Private",
            "value": "private",
          },
          "ref": null,
          "rendered": "Private",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[MockFunction]}
        >
          <option
            value="public"
          >
            Public
          </option>
          <option
            value="private"
          >
            Private
          </option>
        </select>,
        "className": "Select",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="public"
            >
              Public
            </option>,
            <option
              value="private"
            >
              Private
            </option>,
          ],
          "onChange": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "public",
            "nodeType": "host",
            "props": Object {
              "children": "Public",
              "value": "public",
            },
            "ref": null,
            "rendered": "Public",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "private",
            "nodeType": "host",
            "props": Object {
              "children": "Private",
              "value": "private",
            },
            "ref": null,
            "rendered": "Private",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
